build.sbt

```diff
@@ -289,34 +316,54 @@ lazy val unit = project
   .settings(
     sharedSettings,
     publish / skip := true,
-    crossScalaVersions --= scala3,
-    addCompilerPlugin("org.typelevel" %% "kind-projector" % "0.10.3"),
+    Compile / unmanagedSourceDirectories ++= multiScalaDirectories("tests/unit").value,
+    libraryDependencies ++= {
+      if (isScala3.value) List()
+      else List(compilerPlugin("org.typelevel" %% "kind-projector" % "0.10.3"))
+    },
     resolvers += Resolver.bintrayRepo("cibotech", "public"),
     scala212LibraryDependencies(
       List(
         "com.cibo" %% "evilplot" % "0.6.3"
       )
     ),
     libraryDependencies ++= List(
-      "co.fs2" %% "fs2-core" % "2.1.0",
-      "org.scalacheck" %% "scalacheck" % V.scalacheck % Test,
-      "org.scalameta" %% "munit" % V.munit % Test,
-      "org.scalameta" %% "testkit" % V.scalameta % Test
+      "co.fs2" %% "fs2-core" % fs2Version.value,
+      "org.scalameta" %% "munit" % munitVersion.value % Test
     ),
     buildInfoPackage := "tests.cli",
+    buildInfoKeys := Seq[BuildInfoKey](
+      "testsInputClassDirectory" -> (testsInput / Compile / classDirectory).value
+    )
+  )
+  .dependsOn(mdoc, testsInput, tests)
+  .enablePlugins(BuildInfoPlugin, MdocPlugin)
+
+lazy val unitJS = project
+  .in(file("tests/unit-js"))
+  .settings(
+    sharedSettings,
+    publish / skip := true,
+    crossScalaVersions --= scala3,
+    Compile / unmanagedSourceDirectories ++= multiScalaDirectories("tests/unit-js").value,
+    libraryDependencies ++= List(
+      "org.scalameta" %% "munit" % munitVersion.value % Test
+    ),
+    buildInfoPackage := "tests.js",
     buildInfoKeys := Seq[BuildInfoKey](
       "testsInputClassDirectory" -> (testsInput / Compile / classDirectory).value
     ),
     mdocJS := Some(jsdocs)
   )
-  .dependsOn(mdoc, js, testsInput, tests)
+  .dependsOn(mdoc, js, testsInput, tests, unit % "test->test")
```

Comment:
Why is `% "test->test"` needed? This pattern shouldn't be necessary when you have dedicated test projects. Utility testing methods can live in `unit/src/main/scala`. I generally avoid `% "test->test"` where possible